scalableminds/webknossos
# Include Volume Segment List in NML

### URL of deployed dev instance (used for testing):
- https://nmlsegments.webknossos.xyz

### Steps to test:
- Create a volume annotation, brush some cells. In segments tab, name some of the cells
- Click Download. Extract the resulting zip, inspect the NML file. Should have something like this:
```
  <volume id="0" location="data_Volume.zip" name="Volume">
    <segments>
      <segment id="1" name="Excellent Segment" created="1655890347264" anchorPositionX="251" anchorPositionY="234" anchorPositionZ="20" />
      <segment id="2" name="Very Good Segment" created="1655890707250" anchorPositionX="242" anchorPositionY="254" anchorPositionZ="45" />
      <segment id="3" name="Fabulous Segment" created="1655893346122" anchorPositionX="284" anchorPositionY="285" anchorPositionZ="21" />
      <segment id="4" name="This one looks kind of like a dog" created="1655897590305" anchorPositionX="292" anchorPositionY="204" anchorPositionZ="40" />
      <segment id="5" created="1655899024700" anchorPositionX="268" anchorPositionY="296" anchorPositionZ="44" />
      <segment id="6" created="1655899445511" anchorPositionX="229" anchorPositionY="268" anchorPositionZ="44" />
      <segment id="7" name="The Best Segment" created="1655899482076" anchorPositionX="260" anchorPositionY="282" anchorPositionZ="43" />
    </segments>
  </volume>
``` 
- reupload
- segment tab of the uploaded annotation should show the same info as before
- upload an old volume zip, should still work

### Issues:
- fixes #6302
- follow-up issue in wk-libs: https://github.com/scalableminds/webknossos-libs/issues/772

------
